src/frontend/js/hooks/useUserWishlistCourses/index.spec.tsx

```diff
@@ -0,0 +1,149 @@
+import { renderHook, waitFor } from '@testing-library/react';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { IntlProvider } from 'react-intl';
+import fetchMock from 'fetch-mock';
+import { PropsWithChildren } from 'react';
+import { Deferred } from 'utils/test/deferred';
+import * as mockFactories from 'utils/test/factories';
+import { createTestQueryClient } from 'utils/test/createTestQueryClient';
+import JoanieApiProvider from 'data/JoanieApiProvider';
+import BaseSessionProvider from 'data/SessionProvider/BaseSessionProvider';
+import { useUserWishlistCourse, useUserWishlistCourses } from '.';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockFactories
+    .ContextFactory({
+      authentication: {
+        backend: 'fonzie',
+        endpoint: 'https://authentication.test',
+      },
+      joanie_backend: {
+        endpoint: 'https://joanie.test',
+      },
+    })
+    .generate(),
+}));
+
+describe('useWishlistCourse', () => {
+  const wishlistCourse = mockFactories.UserWishlistCourseFactory.generate();
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    fetchMock.restore();
+  });
+
+  const Wrapper = ({ children }: PropsWithChildren) => (
+    <IntlProvider locale="en">
+      <QueryClientProvider client={createTestQueryClient({ user: true })}>
+        <JoanieApiProvider>
+          <BaseSessionProvider>{children}</BaseSessionProvider>
+        </JoanieApiProvider>
+      </QueryClientProvider>
+    </IntlProvider>
+  );
+
+  it('retrieves wishlist cours from id', async () => {
+    const responseDeferred = new Deferred();
+
+    const urlGetWishlistById = `https://joanie.test/api/v1.0/wishlist/${wishlistCourse.id}/`;
+    fetchMock.get(urlGetWishlistById, responseDeferred.promise);
+
+    const { result } = renderHook(() => useUserWishlistCourse(wishlistCourse.id), {
+      wrapper: Wrapper,
+    });
+
+    await waitFor(() => expect(result.current.states.fetching).toBe(true));
+    responseDeferred.resolve(wishlistCourse);
+
+    await waitFor(() => expect(result.current.states.fetching).toBe(false));
```

Comment:
I think it's better to wrap responseDeferred within an act nope?
```suggestion
    await act(async () => {
        responseDeferred.resolve(wishlistCourse);
    });

    expect(result.current.states.fetching).toBe(false);
```

Reply:
I dont totaly understand how it work.
What i think i know is that act aren't necessary if we wait a visual change happening after all async have been resolved.

I may be wrong, the way i choose to do it is: 
first i try to await a visual element.
second, i wrap in a act(), basically when i don't find anything to wait for.

Reply:
after our morning talk.
We `wait..findBy()` after react-testing-library methods like `render` or `userEvent` methods like `await userEvent.click()`.
But `Deferred.resolve()` need to be wrapped in a `await act(async () => { myDeferred.resolve() })`